lib/ws.js

```diff
@@ -29,16 +29,22 @@ class BinanceWS {
         path = (isCombined ? this._combinedBaseUrl : this._baseUrl) + path;
         const ws = new WebSocket(path);
         ws.on('message', (json) => {
-            let response = JSON.parse(json);
-            if (this._beautify) {
-                if (response.stream) {
-                    response.data = this._beautifyResponse(response.data);
-                } else {
-                    response = this._beautifyResponse(response);
-                }
+            try {
+              let response = JSON.parse(json);
+              if (this._beautify) {
+                    if (response.stream) {
+                        response.data = this._beautifyResponse (response.data);
+                    } else {
+                        response = this._beautifyResponse (response);
+                    }
+              }
+               eventHandler(response);
+            }catch (e){
+                console.log("error from parsing body..");
             }
-
-            eventHandler(response);
+        });
+        ws.on('error', (erro) => {
+           console.log(erro);
```

Comment:
Can you remove this line and replace it with

`// node.js EventEmitters will throw and exit if no error listener is registered`

Reply:
ws on error was the exact solution to the bug i reported previously. i tested many times.